core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -173,14 +174,17 @@ public AuthSession tableSession(RESTClient sharedClient, Map<String, String> pro
       sessionCache = newSessionCache(name, properties);
     }
 
+    String oauth2ServerUri = properties.get(OAuth2Properties.OAUTH2_SERVER_URI);
```

Comment:
My rationale here is: two auth sessions for 2 different catalogs, but with the same credential, would be wrongly conflated since they would share the same cache key.

Reply:
isn't that a separate issue from the signer one?